wgpu-core/src/command/bundle.rs

```diff
@@ -819,15 +729,18 @@ fn draw_indexed(
         None => return Err(DrawError::MissingIndexBuffer.into()),
     };
 
-    validate_indexed_draw(
-        &state.vertex[..],
-        &pipeline.steps,
-        index,
-        first_index,
-        index_count,
-        first_instance,
-        instance_count,
-    )?;
+    let vertex_limits = super::VertexLimits::new(state.vertex_buffer_sizes(), &pipeline.steps);
+
+    let last_index = first_index as u64 + index_count as u64;
+    let index_limit = index.limit();
+    if last_index > index_limit {
+        return Err(DrawError::IndexBeyondLimit {
+            last_index,
+            index_limit,
+        }
+        .into());
+    }
```

Comment:
Can you deduplicate this as well?

Reply:
The index limit calculation is simpler to deal with but it's also implemented in two different ways. I'd rather keep this PR focused on the vertex limits.

At some point we should deduplicate all validation code for bundles and passes.